templates/Audio/WAV.tcl

```diff
@@ -43,6 +44,7 @@ proc parse_fmt {} {
 }
 
 proc parse_peak {} {
+    global num_channels
```

Comment:
If you want to skip global var you could possibly use upvar in parse_fmt to a var in parse_list (via upvar in parse_chunk also) and then later pass that var to parse_peak

Reply:
Thanks Mattias, I'm not familiar with Tcl unfortunately but I believe this would introduce quite a bit of indirection if I'm understanding correctly and not sure there's lot to gain from it.

Although I don't like globals either, I hope this one is acceptable for this particular use case (other templates in this repo use globals, too). 👍

Reply:
Yeap a bit more complex. Aha I didn't know global was used in other script, then probably ok. And i guess the tcl interpreter instance is not reused etc

Reply:
Btw, your [fq](https://github.com/wader/fq) is awesome! ✌️

Reply:
Thanks! parts of fq was inspired by hexfiend and the templating, here is an early experiment https://github.com/wader/textfiend :)

After looking at wav.tcl maybe fq's wav decoder needs some shape up 🤔 :)